conn.go

```diff
@@ -405,7 +411,12 @@ func (c *Conn) Write(p []byte) (int, error) {
 // Close closes the connection.
 func (c *Conn) Close() error {
 	err := c.close(true) //nolint:contextcheck
-	c.handshakeLoopsFinished.Wait()
+	c.closeLock.Lock()
```

Comment:
poor `c.closeLock` gets locked and unlocked thrice in a call to `c.Close` (twice in `close` + the call here), I guess I could get the channel in `c.close` and return it? Doesn't fit semantically for `close` to return the channel, but it is an internal function, so it doesn't matter much.